### PR TITLE
Add editor hit animation toggle

### LINKED ISSUE #11022: Sliderends fade out much later than other parts of sliders
Sliderends fade out much later than other parts of sliders in the Editor. Checking by eye, they fade out almost two whole beats later. This makes the editor look clustered when sliders are used in rapid succession. Especially noticeable when approach rate is lower.

Example: [Four sliderends ](https://i.imgur.com/TgC5mXO.jpeg) which have been hit are shown alongside the [following elements](https://i.imgur.com/Whhk91m.jpg).

**osu!lazer version:** 2020.1128.0

### PR DESCRIPTION
**Pull request by peppy**
This brings back a much-requested feature from the `osu-stable` editor, which is that hit circles don't "explode". This is actually the default behaviour in stable, so I've also set it as the default here for the time being, pending feedback.

In addition, I've tweaked the fades and approach circle animation to closer match stable.

https://user-images.githubusercontent.com/191335/115541947-ae8d5400-a2da-11eb-8f78-5caad5ad4454.mp4

Closes https://github.com/ppy/osu/issues/11022.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs [hunk 1]
```diff
@@ -182,6 +182,8 @@ protected override void UpdateHitStateTransforms(ArmedState state)
             // todo: temporary / arbitrary, used for lifetime optimisation.
             this.Delay(800).FadeOut();
 
+            (CirclePiece.Drawable as IMainCirclePiece)?.Animate(state);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs:185]: You may be wondering how this helps in any way. Basically it changes the ordering of execution so the custom logic in `DrawableOsuEditRuleset` runs *after* this is applied. May seem like a hack, but if you look at this change as an isolated one I think it actually feels more elegant than binding to `ApplyCustomUpdateState` in `xCirclePiece`. Open to feedback and discussion, regardless.

File: osu.Game.Rulesets.Osu/Edit/DrawableOsuEditRuleset.cs [hunk 2]
```diff
@@ -58,8 +69,17 @@ private void updateState(DrawableHitObject hitObject, ArmedState state)
...
+                              .FadeOutFromOne(editor_hit_object_fade_out_extension * 4)
                               .Expire();
+
+                        circle.ApproachCircle.ScaleTo(1.1f, 300, Easing.OutQuint);
+
+                        var circlePieceDrawable = circle.CirclePiece.Drawable;
+
+                        // clear any explode animation logic.
+                        circlePieceDrawable.ApplyTransformsAt(circle.HitStateUpdateTime, true);
+                        circlePieceDrawable.ClearTransformsAfter(circle.HitStateUpdateTime, true);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Edit/DrawableOsuEditRuleset.cs:81]: This logic can alternatively be moved inside the `Animate` methods of both `CirclePiece`s and we can call

```csharp
(circlePieceDrawable as IMainCirclePiece)?.Animate(ArmedState.Miss);
```

I had this previously and not sure which I prefer.

**Review by smoogipoo — APPROVED** [review]
Looks reasonable

File: osu.Game.Rulesets.Osu/Edit/DrawableOsuEditRuleset.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Edit/DrawableOsuEditRuleset.cs:81]: I prefer it this way, just to keep these overrides all in one place.

**Pull request merged**